Entities/Common/Building/BlobPlacement.as

```diff
@@ -464,6 +485,19 @@ void onCommand(CBlob@ this, u8 cmd, CBitStream @params)
 			SendGameplayEvent(createBuiltBlobEvent(this.getPlayer(), carryBlob.getName()));
 		}
 	}
+	if (cmd == this.getCommandID("repairBlob"))
+	{
+		CBlob @carryBlob = getBlobByNetworkID(params.read_u16());
+		Vec2f pos = params.read_Vec2f();
+
+		CBlob @repairBlob = getMap().getBlobAtPosition(pos);
```

Comment:
I think it would be better to send the network ID of the repaired blob tile as part of the bitstream rather than figuring out from the serverside. I could imagine it doing weird stuff in some odd cases especially with lag.

e.g. what happens if a 300ms ping builder sends a `repairBlob` command and by the time it's processed serverside, the tile being repaired was broken and _something else entirely_ than a tile now overlaps `pos`? Even if it might be nothing it's making the whole thing more obscure

Reply:
Related to this command, if a laggy builder holds LMB to repair a door, but that door gets broken by the time the command reaches the server, should we make it place a new blob instead?

Reply:
> e.g. what happens if a 300ms ping builder sends a repairBlob command and by the time it's processed serverside, the tile being repaired was broken and something else entirely than a tile now overlaps pos? Even if it might be nothing it's making the whole thing more obscure

would probably end up passing, i've added additional checks in serverBlobCheck now

> Related to this command, if a laggy builder holds LMB to repair a door, but that door gets broken by the time the command reaches the server, should we make it place a new blob instead?

probably should - added that functionality, but either way it requires sending the pos instead of network id 

also, i've added some getHealth() checks along with null checks in certain places because apparently a blob can exist without being null (exist with negative health for a moment and not show up), which meant that if you tried placing a blobtile right after destroying it, it wouldn't place